mqtt-impl/src/main/java/io/streamnative/pulsar/handlers/mqtt/proxy/ProxyHandler.java

```diff
@@ -140,7 +143,17 @@ public void channelRead(ChannelHandlerContext ctx, Object message) throws Except
                     break;
                 case Connected:
                     log.info("channelRead Connected: {}", message);
-                    clientChannel.writeAndFlush(message);
+                    msg = (MqttMessage) message;
+                    messageType = msg.fixedHeader().messageType();
+                    if (messageType == SUBACK) {
+                        MqttSubAckMessage subAckMessage = (MqttSubAckMessage) message;
+                        if (proxyConnection.decreaseSubscribeTopicsCount(
+                                subAckMessage.variableHeader().messageId()) == 0) {
```

Comment:
Maybe the subscription topics count should <= 0?

Reply:
No, we need to ensure only send one response command to the client(Actually this will not happen, send 3 requests to the broker, we will never get more than 3 responses) otherwise the client will close the connection.

Reply:
Ok